src/NuGet.Clients/NuGet.SolutionRestoreManager/NuGet.SolutionRestoreManager.csproj

```diff
@@ -24,6 +24,7 @@
     <ProjectReference Include="..\..\NuGet.Core\NuGet.Protocol\NuGet.Protocol.csproj" />
     <ProjectReference Include="..\..\NuGet.Core\NuGet.Versioning\NuGet.Versioning.csproj" />
     <ProjectReference Include="..\NuGet.PackageManagement.VisualStudio\NuGet.PackageManagement.VisualStudio.csproj" />
+    <ProjectReference Include="..\NuGet.Tools\NuGet.Tools.csproj" />
```

Comment:
This will cause an RPS regression.

NuGet.SolutionRestoreManager is meant to load the minimal amount of assemblies for restore in VS and NuGet.Tools has one of the largest dependency graphs.

Reply:
If I add any ProjectReference to NuGet.SolutionRestoreManager it will cause an RPS regression?

Reply:
NuGet.SolutionRestoreManager is scoped down to the minimal amount of assembly that need to be loaded to kick off a restore at solution load time. 

So yes, adding any new assemblies will cause a regression.